tests/test_auto/test_auto.py

```diff
@@ -72,6 +90,40 @@ def test_objective(
     callback.assert_called_once()
 
 
+@pytest.mark.parametrize("ts_name", ["ts_with_all_folds_missing_one_segment"])
+def test_objective_fail_none(
+    ts_name,
+    request,
+    target_metric=MAE(missing_mode="ignore"),
+    metric_aggregation: Literal["mean"] = "mean",
+    metrics=[MAE(missing_mode="ignore")],
+    backtest_params={},
+    initializer=MagicMock(spec=_Initializer),
+    callback=MagicMock(spec=_Callback),
+    relative_params={
+        "_target_": "etna.pipeline.Pipeline",
+        "horizon": 7,
+        "model": {"_target_": "etna.models.NaiveModel", "lag": 1},
+        "transforms": [{"_target_": "etna.transforms.TimeSeriesImputerTransform"}],
+    },
+):
+    ts = request.getfixturevalue(ts_name)
+    trial = MagicMock(relative_params=relative_params)
+    _objective = Auto.objective(
+        ts=ts,
+        target_metric=target_metric,
+        metric_aggregation=metric_aggregation,
+        metrics=metrics,
+        backtest_params=backtest_params,
+        initializer=initializer,
+        callback=callback,
+    )
+
+    # TODO: discuss the error here
```

Comment:
We should discuss this: it fails before we go into metrics computation.

Reply:
Seems like we should support this behavior now. We can leave it as is for this task and create a separate issue to change this. What do you think?

Reply:
Or we can try to mock here the part that blocks us from testing intended behavior.